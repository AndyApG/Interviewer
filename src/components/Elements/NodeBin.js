import React, { Component } from 'react';

import droppable from '../../behaviors/droppable';

class NodeBin extends Component {
  render() {
    const styles = { position: 'absolute', background: 'green', width: '100px', height: '100px', top: 0, left: 0 }
    return (
      <div className='node-bin' style={ styles } ></div>
    );
  }
}

export default droppable(NodeBin);
