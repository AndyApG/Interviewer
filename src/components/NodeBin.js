import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { compose, withProps } from 'recompose';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { actionCreators as sessionsActions } from '../ducks/modules/sessions';
import { nodePrimaryKeyProperty } from '../ducks/modules/network';
import { DropTarget, MonitorDropTarget } from '../behaviours/DragAndDrop';

/**
  * Renders a droppable NodeBin which accepts `EXISTING_NODE`.
  */
const NodeBin = ({
  willAccept,
  isOver,
}) => {
  const classNames = cx(
    'node-bin',
    { 'node-bin--active': willAccept },
    { 'node-bin--hover': willAccept && isOver },
  );

  return <div className={classNames} />;
};

NodeBin.propTypes = {
  isOver: PropTypes.bool,
  willAccept: PropTypes.bool,
};

NodeBin.defaultProps = {
  isOver: false,
  willAccept: false,
};

function mapDispatchToProps(dispatch) {
  return {
    removeNode: bindActionCreators(sessionsActions.removeNode, dispatch),
  };
}

export default compose(
  connect(null, mapDispatchToProps),
  withProps(props => ({
    accepts: ({ meta }) => meta.itemType === 'EXISTING_NODE',
    onDrop: ({ meta }) => props.removeNode(meta[nodePrimaryKeyProperty]),
  })),
  DropTarget,
  MonitorDropTarget(['isOver', 'willAccept']),
)(NodeBin);
