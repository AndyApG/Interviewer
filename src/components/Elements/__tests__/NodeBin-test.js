/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import NodeBin from '../../Elements/NodeBin';

describe('NodeBin component', () => {
  it('renders ok', () => {
    const component = shallow(<NodeBin />);

    expect(component).toMatchSnapshot();
  });
});
