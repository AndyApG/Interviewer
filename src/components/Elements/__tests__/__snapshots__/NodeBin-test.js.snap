// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeBin component renders ok 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Droppable)
    acceptsDraggableType="EXISTING_NODE"
    dispatch={[Function]}
    droppableName="NODE_BIN"
    isDraggableDeleteable={false}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
  "nodes": Array [
    <Connect(Droppable)
      acceptsDraggableType="EXISTING_NODE"
      dispatch={[Function]}
      droppableName="NODE_BIN"
      isDraggableDeleteable={false}
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(Connect(Droppable))
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
        },
        "dispatchProps": Object {
          "dispatch": [Function],
        },
        "doDispatchPropsDependOnOwnProps": false,
        "doStatePropsDependOnOwnProps": false,
        "finalMapDispatchToProps": [Function],
        "finalMapStateToProps": [Function],
        "hasStoreStateChanged": false,
        "haveOwnPropsChanged": false,
        "haveStatePropsBeenPrecalculated": false,
        "mergedProps": Object {
          "acceptsDraggableType": "EXISTING_NODE",
          "dispatch": [Function],
          "droppableName": "NODE_BIN",
          "isDraggableDeleteable": false,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "props": Object {
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "renderedElement": <Connect(Droppable)
          acceptsDraggableType="EXISTING_NODE"
          dispatch={[Function]}
          droppableName="NODE_BIN"
          isDraggableDeleteable={false}
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
/>,
        "state": Object {
          "storeState": Object {
            "draggable": Object {
              "draggableType": "NODE",
              "isDragging": true,
            },
          },
        },
        "stateProps": Object {
          "acceptsDraggableType": "EXISTING_NODE",
          "droppableName": "NODE_BIN",
          "isDraggableDeleteable": false,
        },
        "statePropsPrecalculationError": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Droppable)
          acceptsDraggableType="EXISTING_NODE"
          dispatch={[Function]}
          droppableName="NODE_BIN"
          isDraggableDeleteable={false}
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <Connect(Droppable)
          acceptsDraggableType="EXISTING_NODE"
          dispatch={[Function]}
          droppableName="NODE_BIN"
          isDraggableDeleteable={false}
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(Connect(Droppable))
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
}
`;
